### ISSUE TITLE
Netapp cdot modules failing

### ISSUE THREAD

**Issue by ieevans** [ansible/ansible#32776]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

storage/netapp/na_cdot_svm

##### ANSIBLE VERSION

```

ansible 2.4.0
ansible 2.5.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/ieevans/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.12 (default, Nov 19 2016, 06:48:10) [GCC 5.4.0 20160609]
```

##### CONFIGURATION

ANSIBLE_NOCOWS(/etc/ansible/ansible.cfg) = True 

##### OS / ENVIRONMENT

Multiple OS's Windows subsystem for linux, Ubunut 16.04.3.
ONTAP 9.2

##### SUMMARY

Trying to use any na_cdot module fails with:

An exception occurred during task execution. To see the full traceback, use -vvv. The error was: netapp_lib.api.zapi.zapi.NaApiError: NetApp API failed. Reason - Unexpected error:
...

##### STEPS TO REPRODUCE

use example na_cdot_svm

```yaml
---
- hosts: localhost

  vars:
    netapp_hostname: "192.168.0.30"
    netapp_username: "ieevans"
    netapp_password: "Netapp123"

  tasks:
    - name: Add SVM
      na_cdot_svm:
        state: present
        name: AnsibleTest
        root_volume: AT_Root
        root_volume_aggregate: aggr0
        root_volume_security_style: unix
        hostname: "{{ netapp_hostname }}"
        username: "{{ netapp_username }}"
        password: "{{ netapp_password }}"
```

##### ACTUAL RESULTS

```
ap -vvv mksvm.yml
ansible-playbook 2.5.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/ieevans/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 2.7.12 (default, Nov 19 2016, 06:48:10) [GCC 5.4.0 20160609]
Using /etc/ansible/ansible.cfg as config file
Parsed /etc/ansible/hosts inventory source with ini plugin

PLAYBOOK: mksvm.yml ******************************************************************************************************************************************************************
1 plays in mksvm.yml

PLAY [localhost] *********************************************************************************************************************************************************************

TASK [Gathering Facts] ***************************************************************************************************************************************************************
task path: /mnt/c/Users/ieeva/Documents/Anisble/NTAP/mksvm.yml:2
Using module file /usr/local/lib/python2.7/dist-packages/ansible/modules/system/setup.py
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: ieevans
<127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/ieevans/.ansible/tmp/ansible-tmp-1510283380.46-100698082508636 `" && echo ansible-tmp-1510283380.46-100698082508636="` echo /home/ieevans/.ansible/tmp/ansible-tmp-1510283380.46-100698082508636 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmpupB4rE TO /home/ieevans/.ansible/tmp/ansible-tmp-1510283380.46-100698082508636/setup.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/ieevans/.ansible/tmp/ansible-tmp-1510283380.46-100698082508636/ /home/ieevans/.ansible/tmp/ansible-tmp-1510283380.46-100698082508636/setup.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /home/ieevans/.ansible/tmp/ansible-tmp-1510283380.46-100698082508636/setup.py; rm -rf "/home/ieevans/.ansible/tmp/ansible-tmp-1510283380.46-100698082508636/" > /dev/null 2>&1 && sleep 0'
ok: [localhost]
META: ran handlers

TASK [Add SVM] ***********************************************************************************************************************************************************************
task path: /mnt/c/Users/ieeva/Documents/Anisble/NTAP/mksvm.yml:10
Using module file /usr/local/lib/python2.7/dist-packages/ansible/modules/storage/netapp/na_cdot_svm.py
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: ieevans
<127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/ieevans/.ansible/tmp/ansible-tmp-1510283383.04-136824392353149 `" && echo ansible-tmp-1510283383.04-136824392353149="` echo /home/ieevans/.ansible/tmp/ansible-tmp-1510283383.04-136824392353149 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmpP7i0IB TO /home/ieevans/.ansible/tmp/ansible-tmp-1510283383.04-136824392353149/na_cdot_svm.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/ieevans/.ansible/tmp/ansible-tmp-1510283383.04-136824392353149/ /home/ieevans/.ansible/tmp/ansible-tmp-1510283383.04-136824392353149/na_cdot_svm.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /home/ieevans/.ansible/tmp/ansible-tmp-1510283383.04-136824392353149/na_cdot_svm.py; rm -rf "/home/ieevans/.ansible/tmp/ansible-tmp-1510283383.04-136824392353149/" > /dev/null 2>&1 && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_enF7lU/ansible_module_na_cdot_svm.py", line 245, in <module>
    main()
  File "/tmp/ansible_enF7lU/ansible_module_na_cdot_svm.py", line 242, in main
    v.apply()
  File "/tmp/ansible_enF7lU/ansible_module_na_cdot_svm.py", line 208, in apply
    vserver_exists = self.get_vserver()
  File "/tmp/ansible_enF7lU/ansible_module_na_cdot_svm.py", line 150, in get_vserver
    enable_tunneling=False)
  File "/usr/local/lib/python2.7/dist-packages/netapp_lib/api/zapi/zapi.py", line 283, in invoke_successfully
    result = self.invoke_elem(na_element, enable_tunneling)
  File "/usr/local/lib/python2.7/dist-packages/netapp_lib/api/zapi/zapi.py", line 265, in invoke_elem
    raise NaApiError('Unexpected error', e.message)
netapp_lib.api.zapi.zapi.NaApiError: NetApp API failed. Reason - Unexpected error:

fatal: [localhost]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_enF7lU/ansible_module_na_cdot_svm.py\", line 245, in <module>\n    main()\n  File \"/tmp/ansible_enF7lU/ansible_module_na_cdot_svm.py\", line 242, in main\n    v.apply()\n  File \"/tmp/ansible_enF7lU/ansible_module_na_cdot_svm.py\", line 208, in apply\n    vserver_exists = self.get_vserver()\n  File \"/tmp/ansible_enF7lU/ansible_module_na_cdot_svm.py\", line 150, in get_vserver\n    enable_tunneling=False)\n  File \"/usr/local/lib/python2.7/dist-packages/netapp_lib/api/zapi/zapi.py\", line 283, in invoke_successfully\n    result = self.invoke_elem(na_element, enable_tunneling)\n  File \"/usr/local/lib/python2.7/dist-packages/netapp_lib/api/zapi/zapi.py\", line 265, in invoke_elem\n    raise NaApiError('Unexpected error', e.message)\nnetapp_lib.api.zapi.zapi.NaApiError: NetApp API failed. Reason - Unexpected error:\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE",
    "rc": 0
}
        to retry, use: --limit @/mnt/c/Users/ieeva/Documents/Anisble/NTAP/mksvm.retry

PLAY RECAP ***************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1
```

**Comment by ansibot** [comment]
cc @broncofan @gouthampacha @hulquest @lmprice
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by gouthampacha** [comment]
@ieevans Do you have HTTP configured and allowed on ONTAP?

**Comment by ieevans** [comment]
Thanks gouthampacha, that was it! 
I had to dig though the ONTAP 9 docs a little to figure out how to get it set.  If anyone else is having the same issue just hit the CLI with admin role and run:
`set -privilege advanced`
Agree to implode the universe 😄, then:
`system services web modify -http-enabled true`

I found the SDK helpful when troubleshooting since the module fails with "Unexpected error:" which wasn't too helpful.

Maybe some notes in the modules docs about enabling HTTP for other users would be helpful. Better yet would be to enable HTTPS somehow. I'm not sure if the NaServer Python module would be required.

Thanks again for all your assistance gouthampacha!
